### PR TITLE
Fem 1677

### PR DESCRIPTION
**Pull request by AntonAFA**
### Description of the Changes

KavaAnalyticsPlugin added. 

### Details

- mediaSource and isLiveStream removed from PLAYBACK_PARAMS_UPDATED. isLiveStream added to Player API
- Video/Audio/Text track changed event added. Each event will hold the new track that have been applied.
- SEEKING event now will hold the requested position.
- KavaAnalyticsPlugin added.

### CODE REVIEW

File: playkit/src/main/java/com/kaltura/playkit/PlayerEvent.java [hunk 1]
```diff
@@ -123,6 +126,46 @@ public Error(PKError error) {
         }
     }
 
+    public static class Seeking extends PlayerEvent {
+
+        public final long requestedPosition;
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/PlayerEvent.java]: targetPosition

File: playkit/src/main/java/com/kaltura/playkit/api/ovp/services/KavaService.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    private static final PKLog log = PKLog.get(KavaService.class.getSimpleName());
+
+    public static RequestBuilder sendAnalyticsEvent(String baseUrl, Map<String, String> params) {
+        return new RequestBuilder()
+                .method("GET")
+                .url(buildUrlWithParams(baseUrl, params));
+    }
+
+    private static String buildUrlWithParams(String baserUrl, Map<String, String> params) {
+        Uri.Builder builder = new Uri.Builder();
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/api/ovp/services/KavaService.java]: Uri.Builder builder = Uri.parse(baseUrl).buildUpon();

File: playkit/src/main/java/com/kaltura/playkit/api/ovp/services/KavaService.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        builder.path(baserUrl);
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/api/ovp/services/KavaService.java]: remove

File: playkit/src/main/java/com/kaltura/playkit/api/ovp/services/KavaService.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+        Iterator iterator = params.entrySet().iterator();
+        Map.Entry<String, String> entry;
+        while (iterator.hasNext()) {
+            entry = (Map.Entry) iterator.next();
+            builder.appendQueryParameter(entry.getKey(), entry.getValue());
+        }
+
+        try {
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/api/ovp/services/KavaService.java]: Check if this is really needed.

File: playkit/src/main/java/com/kaltura/playkit/player/PlayerController.java [hunk 5]
```diff
@@ -115,7 +122,7 @@ public void onEvent(PlayerEvent.Type eventType) {
             //log.d("onEvent event =  " + eventType.name());
             if (eventListener != null) {
 
-                PKEvent event = null;
+                PKEvent event;
 
                 // TODO: use specific event class
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/player/PlayerController.java]: Remove...

File: playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java [hunk 6]
```diff
@@ -74,6 +72,7 @@
     private List<TextTrack> textTracks = new ArrayList<>();
 
     private String[] lastSelectedTrackIds;
+    private String[] requestedChangeTrackIds = {NONE, NONE, NONE};
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java]: targetTrackIds?

File: playkit/src/main/java/com/kaltura/playkit/plugins/ovp/KavaAnalyticsConfig.java [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+    private String ks;
+    private String baseUrl = DEFAULT_BASE_URL;
+    private String customVar1, customVar2, customVar3;
+    private String playbackContext;
+    private String referrerAsBase64;
+
+    public KavaAnalyticsConfig() {}
+
+    public KavaAnalyticsConfig(JsonObject config) {
+        if (Utils.isJsonObjectValueValid(config, UI_CONF_ID_KEY)) {
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/plugins/ovp/KavaAnalyticsConfig.java]: Consider using Gson.

File: playkit/src/main/java/com/kaltura/playkit/plugins/ovp/KavaAnalyticsPlugin.java [hunk 8]
```diff
@@ -0,0 +1,495 @@
...
+
+        params.put("eventType", Integer.toString(event.getValue()));
+        params.put("partnerId", Integer.toString(pluginConfig.getPartnerId()));
+        params.put("entryId", mediaConfig.getMediaEntry().getId());
+        params.put("sessionId", sessionId);
+        params.put("eventIndex", Integer.toString(eventIndex));
+        params.put("referrer", referrer);
+        params.put("deliveryType", deliveryType);
+        params.put("playbackType", getPlaybackType());
+        params.put("clientVer", PlayKitManager.CLIENT_TAG);
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/plugins/ovp/KavaAnalyticsPlugin.java]: Is this needed?

File: playkit/src/main/java/com/kaltura/playkit/plugins/ovp/KavaAnalyticsConfig.java [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+
+    boolean hasUiConfId() {
+        return uiconfId != 0;
+    }
+
+    private boolean isValidReferrer(String referrer) {
+        return (referrer.startsWith("app://") || referrer.startsWith("http://") || referrer.startsWith("https://"));
+    }
+
+    boolean isPartnerIdValid() {
```
Comment (ElizaSapir) [inline playkit/src/main/java/com/kaltura/playkit/plugins/ovp/KavaAnalyticsConfig.java]: @AntonAFA  there is no use of this method

**Pull request merged**